watertap/examples/flowsheets/crystallization/multi_effect_crystallizer.py

```diff
@@ -0,0 +1,306 @@
+import pandas as pd
+import numpy as np
+import pytest
+from pyomo.environ import (
+    ConcreteModel,
+    TerminationCondition,
+    SolverStatus,
+    value,
+    Var,
+)
+from pyomo.network import Port
+from idaes.core import FlowsheetBlock
+from pyomo.util.check_units import assert_units_consistent
+from watertap.unit_models.crystallizer import Crystallization
+import watertap.property_models.cryst_prop_pack as props
+
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+)
+from idaes.core.util.testing import initialization_tester
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    unscaled_variables_generator,
+    badly_scaled_var_generator,
+)
+from idaes.core import UnitModelCostingBlock
+
+from watertap.costing import WaterTAPCosting, CrystallizerCostType
+
+solver = get_solver()
+
+
+def build_multi_effect_crystallizer_fs(
+    m=None,
+    operating_pressure_eff1=0.35,  # bar
+    operating_pressure_eff2=0.208,  # bar
+    operating_pressure_eff3=0.095,  # bar
+    feed_mass_frac_NaCl=0.2126,
+    feed_pressure=101325,  # Pa
+    feed_temperature=273.15 + 20,  # K
+    crystallizer_yield=0.5,
+):
+    """
+    This flowsheet depicts a 3-effect crystallizer, with brine fed in parallel
+    to each effect, and the operating pressure is specfied individually.
+    """
+
+    if m is None:
+        m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+    m.fs.props = props.NaClParameterBlock()
+
+    eff_1 = m.fs.effect_1 = Crystallization(property_package=m.fs.props)
+    eff_2 = m.fs.effect_2 = Crystallization(property_package=m.fs.props)
+    eff_3 = m.fs.effect_3 = Crystallization(property_package=m.fs.props)
```

Comment:
So there is no explicit addition of IDAES heat exchangers in the flowsheet. Are you considering any efficiency losses, and is there a reason not to explicitly add HXs?

Reply:
Initially I also wanted to include the IDAES heat exchanger (0D) model to consider transfer loss, but I didn't see it's included in the model - could be wrong about that. This PR focuses on determining how much waste water could be  disposed in each effect, so knowing the energy flow is sufficient, but of course the feed flow will decrease a bit when considering the losses.

However, I do plan to include the IDAES HX model (and a self-implemented costing function) for the cost optimization, as its capital cost (heat transfer area) is dependent on the approaching temperature between effects. Hope it makes sense.